client/src/main/scala/org/http4s/client/DefaultClient.scala

```diff
@@ -39,18 +39,18 @@ private[http4s] abstract class DefaultClient[F[_]](implicit F: Bracket[F, Throwa
     * @return The result of applying f to the response to req
     */
   def fetch[A](req: F[Request[F]])(f: Response[F] => F[A]): F[A] =
-    req.flatMap(fetch(_)(f))
+    req.flatMap(run(_).use(f))
 
   /**
     * Returns this client as a [[Kleisli]].  All connections created by this
     * service are disposed on completion of callback task f.
     *
-    * This method effectively reverses the arguments to `fetch`, and is
+    * This method effectively reverses the arguments to `run` followed by `use`, and is
     * preferred when an HTTP client is composed into a larger Kleisli function,
     * or when a common response callback is used by many call sites.
     */
   def toKleisli[A](f: Response[F] => F[A]): Kleisli[F, Request[F], A] =
-    Kleisli(fetch(_)(f))
+    Kleisli(run(_).use(f))
```

Comment:
And maybe deprecate this one?  It could go to `Kleisli[Resource[F, ?], Request[F], Response[F]]`, but maybe it's not useful.

Reply:
I see there're many legacy stuff to deprecate in `Client`. I think I can take on it too. But I'd prefer to work on it in a separate PR, just to keep this one maintainable.